feat: homepage styles

cuzuco2 opened the pull request:
Changes for the dak-light style on pages

Fix #8

Test URLs:
- Before: https://main--elco-aerotop-sg--hlxsites.hlx.page/de/
- After: https://dark-light-style--elco-aerotop-sg--hlxsites.hlx.page/de/

Commit:
First changes on backgriund and text color for dark block

Commit:
Adding style for dark mode

cuzuco2 commented:
@ekremney @solaris007 Please review

solaris007 commented:
please fix linting error:
```
styles/styles.css
 175:1  ✖  Expected selector "a:any-link" to come before selector "body       no-descending-specificity
           .dark a"
 180:1  ✖  Expected selector "a:hover" to come before selector "body .dark    no-descending-specificity
           a"
```

solaris007 committed:
Merge branch 'main' into dark-light-style

solaris007 commented on styles/styles.css:
```diff
@@ -120,6 +120,15 @@ body.appear {
   display: unset;
 }
```
please fix linting error:
```
styles/styles.css
 175:1  ✖  Expected selector "a:any-link" to come before selector "body       no-descending-specificity
           .dark a"
 180:1  ✖  Expected selector "a:hover" to come before selector "body .dark    no-descending-specificity
           a"
```

cuzuco2 replied:
done

solaris007 commented:
The top of the dark area is too low. Probably a general issue on all pages. Please fix as part of this PR.

![Screenshot 2023-01-17 at 08 09 45](https://user-images.githubusercontent.com/1872195/212832652-62fcdef6-1823-45ae-b170-07fa14d07b97.png)

solaris007 commented:
@cuzuco2 i allowed myself to include a couple of additional homepage-related things in the linked issue #8.

- Implement dark vs light styling using section metadata (see example on German `index.docx`)
- All images have a in/out transition (something like: `    transform: matrix3d(1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1);`)
- Margins/Paddings
- Font sizes
- Colors
- Remove auto-transform of links on entire site; style links on homepage as buttons

solaris007 commented on styles/styles.css:
```diff
@@ -120,6 +120,15 @@ body.appear {
   display: unset;
 }
 
+body .dark {
```
The homepage content, mostly in the dark areas seems to center content. Can you investigate?

cuzuco2 replied:
@solaris007 sometimes it is centered sometimes it is not:
![image](https://user-images.githubusercontent.com/44608307/212936612-e5f02f48-ba3d-4e03-8a59-4b53b662204c.png)
![image](https://user-images.githubusercontent.com/44608307/212936840-e249e19e-3658-4ab9-9a6f-a61562eb803a.png)
I understand that we cannot add just a metadata for such thing, how would you like to approach this? with a new block?

cuzuco2 replied:
Done, I added a new metadata to the section.

solaris007 committed:
Merge branch 'main' into dark-light-style

Commit:
Adding styling to tak einto account the space on the first section on top

Commit:
Merging with the changes on sytling  for the main page
Merge branch 'dark-light-style'  f github.com:hlxsites/elco-aerotop-sg into dark-light-style

solaris007 committed:
Merge branch 'main' into dark-light-style

Commit:
GEtting the latest updates
Merge branch 'dark-light-style' of github.com:hlxsites/elco-aerotop-sg into dark-light-style

Commit:
Changes to account for the different padding and borders between dark and light sections

Commit:
Fixed lint issues

cuzuco2 commented:
@solaris007 @ekremney Please review now, it should address all the different elements except for the image transition which has it's own separate issue.
Also this doesn't deal with image floating for now, or images with columns with title and descriptions as seen in here:
![image](https://user-images.githubusercontent.com/44608307/212950621-d24dbdef-5c09-420e-a2be-6af45d03dbba.png)

Commit:
Fixed centered ot non centered text on blocks

ekremney commented on styles/styles.css:
```diff
@@ -338,3 +337,68 @@ main .section[data-section-status='initialized'] {
+  padding-bottom: 65px;    
+}
+
+.section.dark.center {
```
_nitpicking_: please fix the indentation for this and the following blocks

cuzuco2 replied:
done

solaris007 commented on styles/styles.css:
```diff
@@ -338,3 +337,68 @@ main .section[data-section-status='initialized'] {
   background-color: var(--highlight-background-color);
 }
+
+body > main > div> div {
```
i would be more specific here and apply to `section`

cuzuco2 replied:
done

solaris007 commented on styles/styles.css:
```diff
@@ -338,3 +337,68 @@ main .section[data-section-status='initialized'] {
+  margin-bottom: 0;
+}
+
+body > main > div:first-child > div {
```
apply to first `section`

solaris007 commented:
will you handle this item of issue #8 separately?

All images have a in/out transition (something like:     transform: matrix3d(1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1);)

cuzuco2 commented:
@solaris007 , @ekremney  changed the code according to your review. I will handle the transition separately as the transition has it's own issue so like we said we will create a different branch and different PR for each different issue.
@solaris, if it is fine with you please merge the pull request

Commit:
Fixed base donr eview indentation and context of css
